src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const item = this[i];
+
+      if (item !== null && item !== undefined) {
+        str += '' + separator + item;
+      } else {
+        str += separator;
+      }
+    }
+
+    if (separator !== '') {
+      const sep = `${separator}`;
+
+      str = str.substring(sep.length);
+    }
```

Comment:
let's change the logic a bit to make the code simpler. two plane ifs will be enough. decide if we need to add an element if the first one, and if we need to add the separator in the second